### ISSUE TITLE
assemble with only the forward strand - don't consider RCs

### ISSUE THREAD

**Issue by conchoecia** [marbl/canu#804]
Let's say that I have some prior information where I know that all of my reads are in the correct orientation for OLC. This means that we do not need to consider the RC of the reads when running `canu`. Is it possible to constrain `canu` in such a way? 

If my question isn't clear I am happy to try to explain it better. Thanks!

**Comment by brianwalenz** [comment]
I can think of a couple of ways that I could do it by modifying the code, but I won't do that quite yet.  It'd be possible for an advanced user to do it - but by rewriting the overlaps to remove any false overlaps - but you'd still _generate_ the overlaps.

What's the use case?  Is this a speed optimization or a quality optimization?  Are these real reads (and so need to be corrected too) or high quality reads from some oracle?

**Comment by conchoecia** [comment]
This is a quality optimization - for this application I already have an oracle that tells us the expected general order and orientation of the reads. These are with real reads. The goal is to help `canu` out by telling it how it should try to generate the overlaps.

Thinking about it more, another option is just to check the `canu` contig and see if they match what our oracle says.
 
I can't get too much into the specifics but it sounds like I'll have to go back to the drawing board about how I want to use `canu`!

**Comment by brianwalenz** [comment]
I can probably hack in a solution relatively easily, but it'll be a compile time option.  When loading overlaps into the store, I can just simply discard those that aren't normally oriented (both forward or both reverse).  Want it?

**Comment by conchoecia** [comment]
That would be wonderful if you have the time. It won't be too difficult for me to just keep two separate installs. Thanks for willing to try it out!

**Comment by brianwalenz** [comment]
It's there!  In src/stores/ovStoreFilter.C, change the 0 to 1 on line 167 to enable (context should be obvious).  The overlaps are still computed as before, this just marks all overlaps between a forward and a complement read as 'bad', so only fwd-fwd (or rev-rev) overlaps are loaded into the store.

I counted the number of such overlaps removed, but canu helpfully removes the log.  Comment out the unlink() in src/pipelines/canu/OverlapStore.pm on line 108 to keep it.

```
-- BUCKETIZING --

-  Bucketizing '1-overlapper/001/000001.ovb'
-- Create bucket './test.ovlStore.BUILDING/tmp.sort.0001'
-  Bucketizing '1-overlapper/001/000002.ovb'
-  Bucketizing '1-overlapper/001/000003.ovb'
-  Bucketizing finished:
-- Saved      432530 unitigging overlaps
-- Discarded  437086 opposite orientation
```

I ran e.coli and got two complete contigs.  Two for one!

![ecoli](https://user-images.githubusercontent.com/8249720/37176821-7b239970-22eb-11e8-94da-7a4bd5de6309.png)
